aspnetcore/grpc/performance.md

```diff
@@ -70,8 +70,43 @@ There are a couple of workarounds for .NET Core 3.1 apps:
 > * Thread contention between streams trying to write to the connection.
 > * Connection packet loss causes all calls to be blocked at the TCP layer.
 
+## Load balancing
+
+Load balancing gRPC calls may not work well in some environments by default. L4 (transport) load balancers operate at a connection level, by distributing TCP connections across endpoints. This approach works well for loading balancing API calls made with HTTP/1.1. Concurrent calls made with HTTP/1.1 are sent on different connections, allowing calls to be load balanced across endpoints.
+
+L4 load balancing doesn't work well with gRPC. gRPC uses HTTP/2, which multiplexes multiple calls on a single TCP connection. Because there is a single connection, all calls go to one endpoint.
+
+There are two options to effectively load balance gRPC:
+
+1. Client-side load balancing
+2. L7 (application) proxy load balancing
+
+### Client-side load balancing
+
+With client-side load balancing, the client knows about endpoints. For each gRPC call it selects a different endpoint to send the call to. Client-side load balancing is a good choice for when latency is important. There is no proxy between the client and the service so the call is sent to the service directly. The downside to client-side load balancing is that each client must keep track of available endpoints it should use.
+
+Lookaside client load balancing is a technique where load balancing state is stored in a central location. Clients periodically query the central location for information to use when making load balancing decisions.
+
+`Grpc.Net.Client` currently doesn't support client-side load balancing. The [Grpc.Core](https://www.nuget.org/packages/Grpc.Core) is a good choice if client-side load balancing is required in .NET.
+
+### Proxy load balancing
+
+An L7 (application) proxy works at a higher level than an L4 (transport) proxy. L7 proxies understand HTTP/2, and are able to distribute gRPC calls multiplexed to the proxy on one HTTP/2 connection across multiple endpoints. Using a proxy is simpler than client-side load balancing, but can add extra latency to gRPC calls.
```

Comment:
Note even an L7 proxy will usually only be able to load balance distinct requests. gRPC messages streamed on a single request would not be load balanced.

Reply:
That's true. At least in this doc I don't want to go into that level of detail.